core/ir/x86/encode.c

```diff
@@ -1284,17 +1284,7 @@ opnd_type_ok(decode_info_t *di /*prefixes field is IN/OUT; x86_mode is IN*/, opn
         return (opnd_is_far_pc(opnd) || opnd_is_far_instr(opnd));
     }
     case TYPE_O:
-        return ((opnd_is_abs_addr(opnd) ||
-#ifdef X64
-                 /* We'll take a relative address that rip-rel won't reach:
-                  * after all, OPND_CREATE_ABSMEM() makes a rip-rel.
-                  */
-                 (opnd_is_rel_addr(opnd) &&
-                  (!REL32_REACHABLE(di->final_pc + MAX_INSTR_LENGTH,
```

Comment:
Hmm removing this completely was not what I had in mind initially.  Also I wanted to change at least one place where a test deliberately uses a PC of 0 as an additional fix for this issue.  Will try to get more feedback in next day.  Thank for taking a stab at this; it was on my list but never made it to the top.